### PR TITLE
Avoid deprecated `is_categorical_dtype` from `pandas`

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/backends.py
Comment (jrbourbeau) [inline dask/dataframe/backends.py:693]: Nitpick (this is shorter and more consistent with what we're doing elsewhere)

```suggestion
    return isinstance(dtype, pd.CategoricalDtype)
```

File: dask/dataframe/core.py
Comment (jrbourbeau) [inline dask/dataframe/core.py:3423]: Why do we need `pd.api.types.pandas_dtype` here?

**Review by jrbourbeau — COMMENTED** [review]
Thanks @phofl 

xref https://github.com/pandas-dev/pandas/pull/52527

File: dask/dataframe/backends.py
Comment (phofl) [reply dask/dataframe/backends.py:693]: Yeah this confused me as well (we have a similar ExtensionDtype above) will adjust

File: dask/dataframe/core.py
Comment (phofl) [reply dask/dataframe/core.py:3423]: We can get „category“ as string instead of a dtype object here,pandas_dtype takes care of the proper conversion

Comment (jrbourbeau) [reply dask/dataframe/core.py:3423]: Ah, I see. Thanks for clarifying 👍

**Comment by phofl** [comment]
Failure look unrelated, but would like to get the other deprecation pr in before

File: dask/dataframe/core.py
Comment (j-bennet) [reply dask/dataframe/core.py:3423]: This would work with string or dtype, right?

```python
if pd.api.types.is_dtype_equal(v, pd.CategoricalDtype())
```

File: dask/dataframe/backends.py
Comment (phofl) [reply dask/dataframe/backends.py:406]: It's faster, by a lot. Doesn't matter if you are calling it only once, but since it's not more complex I wouldn't use ``is_dtype_equal`` if possible.  (can probably count the occasions I used it on one hand)

```
dtype = pd.CategoricalDtype()
%timeit isinstance(dtype, pd.CategoricalDtype)
48.5 ns ± 0.163 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
%timeit pd.api.types.is_dtype_equal(dtype, pd.CategoricalDtype())
1.44 µs ± 3.67 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

File: dask/dataframe/core.py
Comment (phofl) [reply dask/dataframe/core.py:3423]: Yes, generally I prefer ``pandas_dtype`` though since it raises if you provide something invalid

File: dask/dataframe/tests/test_categorical.py
Comment (jrbourbeau) [inline dask/dataframe/tests/test_categorical.py:195]: IIUC I think this test can actually be removed. Even prior to this PR, it looks like it was just testing `pandas` `is_categorical_dtype` method. This was added a while ago in https://github.com/dask/dask/pull/1602 where we had some extra logic around where `is_categorical_dtype` needed to be imported from `pandas`. Since we don't have that anymore, we should be good to remove this test

**Review by jrbourbeau — COMMENTED** [review]
Thanks @phofl -- just one final comment

**Review by jrbourbeau — APPROVED** [review]
Thanks @phofl. Will merge after CI is finished (had to resolve a small merge conflict)

**Pull request merged**
